packages/react-instantsearch-dom/src/core/__tests__/createInstantSearchServer.js

```diff
@@ -69,9 +65,9 @@ describe('createInstantSearchServer', () => {
         searchClient,
       };
 
-      shallow(<InstantSearch {...props} />);
+      mount(<InstantSearch {...props} />);
 
-      expect(searchClient.addAlgoliaAgent).toHaveBeenCalledTimes(2);
+      expect(searchClient.addAlgoliaAgent).toHaveBeenCalledTimes(4);
```

Comment:
Why is it called four times?

Reply:
`<InstantSearch>`'s constructor will call `createInstantSearchManager` and `createInstantSearchManager`'s constructor will call `algoliasearchHelper`.

* `algoliasearchHelper`'s constructor calls it **one time**  with `JS Helper {version}`
* `algoliasearchHelper`'s constructor calls it a **second time**  (through `this.setClient`) with `JS Helper {version}` (not sure if this is wanted but it is the way it is done in the helper)
* `createInstantSearchManager`'s constructor calls `addAlgoliaAgents` that will call it **two more times**  with `react ({version})` and `react-instantsearch ({version})`

Before this change only the calls by `algoliasearchHelper` were made.

I agree that this is counter intuitive, a comment could be added here to explain why it has been called 4 times